Sources/Moya/Plugins/AccessTokenPlugin.swift

```diff
@@ -13,15 +13,27 @@ public protocol AccessTokenAuthorizable {
 // MARK: - AuthorizationType
 
 /// An enum representing the header to use with an `AccessTokenPlugin`
-public enum AuthorizationType: String {
+public enum AuthorizationType {
     /// No header.
     case none
 
     /// The `"Basic"` header.
-    case basic = "Basic"
+    case basic
 
     /// The `"Bearer"` header.
-    case bearer = "Bearer"
+    case bearer
+    
+    /// Custom header implementation.
+    case custom(String)
+    
+    public var value: String {
```

Comment:
Scratch that, `ValidationType` returns an empty array for `.none` so I think this is fine as well. Better to remain consistent

Reply:
I think that returning optional value is good though. Right now you need to have the knowledge that the `value` is useless for `.none` case. With optional value you'll always get a meaningful value from it, if you don't, you'll get `nil`.

It is kinda different with `ValidationType`, because you'll either way check if the `statusCode` is in the returning array. In this case you will need an _additional conditional statement_, that if the string is empty, do not do anything.
But yeah we could improve the `ValidationType` in the future if needed.

Reply:
That’s a great point! You rarely if ever see optional arrays for that reason. I change my mind again 😆